traitsui/editors/shell_editor.py

```diff
@@ -50,16 +50,21 @@ def init(self, parent):
         value = self.value
         if self.factory.share and isinstance(value, dict):
             locals = value
-        self._shell = shell = PythonShell(parent, locals=locals)
```

Comment:
`PythonShell` has no `locals` trait, nor could I find anything to do with `locals` in an `__init__` method in the class hierarchy.  All `locals` related code runs through `shell.interpreter().locals` AFAICT. 
I am a little wary of this change, but the previous code did not make sense to me...

I will try to think of a test to see that this was indeed a bug

Reply:
ref:
https://github.com/enthought/pyface/blob/master/pyface/i_python_shell.py
https://github.com/enthought/pyface/blob/master/pyface/ui/qt4/python_shell.py
https://github.com/enthought/pyface/blob/master/pyface/ui/wx/python_shell.py

Reply:
Agreed. I don't see any object in the heirarchy making use of the `locals` kwarg